### PR TITLE
[zh-cn]: update the translation of async-function in statement section

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

As the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#18266

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+  sourceCommit: b15c892914bbaa51a8c8f288426ebc6d5c1d57f8
 ---
 
 {{jsSidebar("Statements")}}
 
-async 函数是使用`async`关键字声明的函数。async 函数是 {{jsxref("Global_Objects/AsyncFunction","AsyncFunction")}} 构造函数的实例，并且其中允许使用 `await` 关键字。`async` 和 `await` 关键字让我们可以用一种更简洁的方式写出基于 {{jsxref("Promise")}} 的异步行为，而无需刻意地链式调用 `promise`。
+**`async function`** 声明创建一个{{Glossary("binding", "绑定")}}到给定名称的新异步函数。函数体内允许使用 `await` 关键字，这使得我们可以更简洁地编写基于 promise 的异步代码，并且避免了显式地配置 promise 链的需要。
 
-async 函数还可以被{{jsxref("Operators/async_function", "作为表达式", "", 1)}}来定义。
+你也可以使用 [`async function` 表达式](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function) 来定义异步函数。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:12]:
```suggestion
你也可以使用 [`async function` 表达式](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function)来定义异步函数。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 2]
```diff
@@ -20,51 +22,78 @@ async function name(param0) {
...
-### 返回值
-
-一个 {{jsxref("Promise")}}，这个 promise 要么会通过一个由 async 函数返回的值被解决，要么会通过一个从 async 函数中抛出的（或其中没有被捕获到的）异常被拒绝。
-
 ## 描述
 
-async 函数可能包含 0 个或者多个 {{jsxref("Operators/await", "await")}} 表达式。await 表达式会暂停整个 async 函数的执行进程并出让其控制权，只有当其等待的基于 promise 的异步操作被兑现或被拒绝之后才会恢复进程。promise 的解决值会被当作该 await 表达式的返回值。使用 `async`/`await` 关键字就可以在异步代码中使用普通的 `try`/`catch` 代码块。
+`async function` 声明创建一个 {{jsxref("AsyncFunction")}} 对象。每次调用异步函数时，都会返回一个新的 {{jsxref("Promise")}} 对象，该对象将会被解决为异步函数的返回值，或者被拒绝为异步函数中未捕获的异常。
+
+异步函数可以包含零个或者多个 {{jsxref("Operators/await", "await")}} 表达式。await 表达式通过暂停执行使返回 promise 的函数函数表现得像同步函数一样，直到返回的 promise 被兑现或拒绝。返回的 promise 的解决值会被当作该 await 表达式的返回值。使用 `async`/`await` 关键字就可以使用普通的 `try`/`catch` 代码块捕获异步代码中的错误。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:45]:
```suggestion
异步函数可以包含零个或者多个 {{jsxref("Operators/await", "await")}} 表达式。await 表达式通过暂停执行使返回 promise 的函数表现得像同步函数一样，直到返回的 promise 被兑现或拒绝。返回的 promise 的解决值会被当作该 await 表达式的返回值。使用 `async`/`await` 关键字就可以使用普通的 `try`/`catch` 代码块捕获异步代码中的错误。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 3]
```diff
@@ -102,134 +131,142 @@ async function foo() {
...
   const p1 = new Promise((resolve) => setTimeout(() => resolve("1"), 1000));
   const p2 = new Promise((_, reject) => setTimeout(() => reject("2"), 500));
-  const results = [await p1, await p2]; // 不推荐使用这种方式，请使用 Promise.all 或者 Promise.allSettled
+  const results = [await p1, await p2]; // 不要这么写！请使用 Promise.all 或者 Promise.allSettled。
 }
-foo().catch(() => {}); // 捕捉所有的错误...
+foo().catch(() => {}); // 尝试捕捉所有的错误...
 ```
 
+`async function` 声明的行为类似于 {{jsxref("Statements/function", "function")}} 声明，它会被[提升](/zh-CN/docs/Glossary/Hoisting)到其作用域的顶部，并且可以在其作用域的任何位置被调用，并且只能在特定情况下被重新声明。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:147]:
```suggestion
`async function` 声明的行为类似于 {{jsxref("Statements/function", "function")}} 声明，它会被[提升](/zh-CN/docs/Glossary/Hoisting)到其作用域的顶部，并且可以在其作用域的任何位置被调用，并且只能在其他上下文中被重新声明。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 4]
```diff
@@ -102,134 +131,142 @@ async function foo() {
...
-在 `concurrentStart` 中，两个计时器被同时创建，然后执行 `await`。这两个计时器同时运行，这意味着程序完成运行只需要 2 秒，而不是 3 秒，即最慢的计时器的时间。
-
-但是 `await` 仍旧是顺序执行的，第二个 `await` 还是得等待第一个执行完。在这个例子中，这使得先运行结束的输出出现在最慢的输出之后。
+#### await 和并发执行
 
-如果你希望并行执行两个或更多的任务，你必须像在`parallel`中一样使用`await Promise.all([job1(), job2()])`。
+在 `sequentialStart` 中，程序执行第一个 `await` 时暂停 2 秒，然后又为第二个 `await` 暂停了 1 秒。直到第一个计时器结束后，第二个计时器才被创建，因此程序需要 3 秒执行完毕。
 
-#### async/await 和 Promise/then 对比以及错误处理
+在 `sequentialWait` 中，两个计时器都被创建并用 `await` 进行等待。这两个计时器并行运行，这意味着代码运行时间缩短到 2 秒，而不是 3 秒，即最慢的计时器的时间。然而，`await` 调用仍旧是顺序执行的，这意味着第二个 `await` 会等待第一个执行完。在这个例子中，较快的计时器的结果会在较慢的计时器之后被处理。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:245]:
```suggestion
在 `sequentialWait` 中，两个计时器都被创建并用 `await` 进行等待。这两个计时器并行运行，这意味着代码运行时间缩短到 2 秒，而不是 3 秒，即较慢的计时器的时间。然而，`await` 调用仍旧是顺序执行的，这意味着第二个 `await` 会等待第一个执行完。在这个例子中，较快的计时器的结果会在较慢的计时器之后被处理。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 5]
```diff
@@ -102,134 +131,142 @@ async function foo() {
...
 
-大多数 async 函数也可以使用 Promises 编写。但是，在错误处理方面，async 函数更容易捕获异常错误
-
-上面例子中的`concurrentStart`函数和`concurrentPromise`函数在功能上都是等效的。在`concurrentStart`函数中，如果任一`await`ed 调用失败，它将自动捕获异常，async 函数执行中断，并通过隐式返回 Promise 将错误传递给调用者。
+在 `concurrentStart` 中，两个计时器被同时创建，然后执行 `await`。这两个计时器同时运行，这意味着程序完成运行只需要 2 秒，而不是 3 秒，即最慢的计时器的时间。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:247]:
```suggestion
在 `concurrentStart` 中，两个计时器被同时创建，然后执行 `await`。这两个计时器同时运行，这意味着程序完成运行只需要 2 秒，而不是 3 秒，即较慢的计时器的时间。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 6]
```diff
@@ -243,29 +280,16 @@ async function getProcessedData(url) {
...
-  } catch (e) {
-    return null;
-  }
+  const v = await downloadData(url).catch((e) => downloadFallbackData(url));
+  return processDataInWorker(v);
 }
 ```
 
-简单地写上`return processDataInworker(v);` 将导致在 `processDataInWorker(v)` 出错时 function 返回值为{{jsxref("Promise")}}而不是返回 `null`。`return foo;` 和 `return await foo;`，有一些细微的差异：`return foo;`不管`foo`是 promise 还是 rejects 都将会直接返回`foo`。相反地，如果`foo`是一个{{jsxref("Promise")}}，`return await foo;`将等待`foo`执行 (resolve) 或拒绝 (reject)，如果是拒绝，将会在返回前抛出异常。
+以上两个重写版本中，请注意在 `return` 关键字之后没有 `await` 语句，尽管这也是有效的：异步函数的返回值隐式地被包装在 {{jsxref("Promise.resolve")}} 中——如果它本身不是一个 promise 的话（如上面的例子）。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:292]:
```suggestion
以上两个重写版本中，请注意在 `return` 关键字之后没有 `await` 语句，尽管这也是有效的：异步函数的返回值隐式地被包装在 {{jsxref("Promise.resolve")}} 中——如果它本身不是一个 promise 的话（正如上面的示例所示）。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 7]
```diff
@@ -74,21 +103,21 @@ async function foo() {
...
-在 await 表达式之后的代码可以被认为是存在在链式调用的 then 回调中，多个 await 表达式都将加入链式调用的 then 回调中，返回值将作为最后一个 then 回调的返回值。
+每个 await 表达式之后的代码可以被认为存在于 `.then` 回调中。通过这种方式，可以通过函数的每个可重入步骤来逐步构建 promise 链。而返回值构成了链中的最后一个环。
 
-在接下来的例子中，我们将使用 await 执行两次 promise，整个 `foo` 函数的执行将会被分为三个阶段。
+在接下来的示例中，我们依次 await 两个 promise，整个 `foo` 函数的执行将会被分为三个阶段。
 
-1. `foo` 函数的第一行将会同步执行，await 将会等待 promise 的结束。然后暂停通过 `foo` 的进程，并将控制权交还给调用 `foo` 的函数。
-2. 一段时间后，当第一个 promise 完结的时候，控制权将重新回到 foo 函数内。示例中将会将`1`（promise 状态为 fulfilled）作为结果返回给 await 表达式的左边即 `result1`。接下来函数会继续进行，到达第二个 await 区域，此时 `foo` 函数的进程将再次被暂停。
-3. 一段时间后，同样当第二个 promise 完结的时候，`result2` 将被赋值为 `2`，之后函数将会正常同步执行，将默认返回`undefined` 。
+1. `foo` 函数的第一行将会同步执行，其中 await 配置了待定的 promise。然后 `foo` 的进程将被暂停，并将控制权交还给调用`foo` 的函数。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:118]:
```suggestion
1. `foo` 函数的第一行将会同步执行，其中 await 配置了待定的 promise。然后 `foo` 的进程将被暂停，并将控制权交还给调用 `foo` 的函数。
```

File: files/zh-cn/web/javascript/reference/statements/async_function/index.md [hunk 8]
```diff
@@ -277,7 +301,14 @@ async function getProcessedData(url) {
...
+- [函数](/zh-CN/docs/Web/JavaScript/Reference/Functions)
+- {{jsxref("AsyncFunction")}}
+- [`async function` 表达式](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function)
+- {{jsxref("Statements/function", "function")}}
+- {{jsxref("Statements/function*", "function*")}}
+- {{jsxref("Statements/async_function*", "async function*")}}
 - {{jsxref("Operators/await", "await")}}
-- ["Decorating Async Javascript Functions" on "innolitics.com"](http://innolitics.com/10x/javascript-decorators-for-promise-returning-functions/)
+- {{jsxref("Promise")}}
+- [Decorating async JavaScript functions](https://innolitics.com/10x/javascript-decorators-for-promise-returning-functions/) 发表于 innolitics.com（2016）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/async_function/index.md:314]: Let's also translate this:

```suggestion
- [装饰 async JavaScript 函数](https://innolitics.com/10x/javascript-decorators-for-promise-returning-functions/)，发表于 innolitics.com（2016）
```

**Review by yin1999 — APPROVED** [review]
LGTM. Just two nits:

**Pull request merged**